### PR TITLE
Compass

### PR DESCRIPTION
**Pull request by magangloff**
- Refactored the camera-preview
- Added compass with compass-view on camera-preview
- PR only for demo

TODO:
- Handle compass for different phone orientation (portrait, landscape)
- Make compass-view adapt better for different screen sizes
- Adapt compass-view with the fieldof view of the camera

### CODE REVIEW

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+     * Calculate the aspect ratio of the display in function of the width and height of the screen
+     * @param width width of the preview in Pixels
+     * @param height height of the preview in Pixels
+     * @return Aspect ratio of the phone
+     */
+    private int aspectRatio(int width, int height){
+        double previewRatio = (double)Math.max(width, height)/Math.min(width, height);
+        double RATIO_16_9_VALUE = 16.0 / 9.0;
+        double RATIO_4_3_VALUE = 4.0 / 3.0;
+        if(Math.abs(previewRatio - RATIO_4_3_VALUE) <= Math.abs(previewRatio - RATIO_16_9_VALUE)){
```
Comment (giommok) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java:137]:
```suggestion
        return (Math.abs(previewRatio - RATIO_4_3_VALUE) <= Math.abs(previewRatio - RATIO_16_9_VALUE)) ? RATIO_4_3 : RATIO_16_9;
```
This way we could write all the remaining code of this method in just one line.
Everything else fine!

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            return AspectRatio.RATIO_4_3;
```
Comment (giommok) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java:138]:
```suggestion
```

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        }
```
Comment (giommok) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java:139]:
```suggestion
```

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        return AspectRatio.RATIO_16_9;
```
Comment (giommok) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java:140]:
```suggestion
```

**Review by giommok — CHANGES_REQUESTED** [review]

**Review by bgabriel998 — APPROVED** [review]
Approve changes suggested by Giovanni

**Review by giommok — APPROVED** [review]

**Pull request merged**